src/plugins/lua/weechat-lua.c

```diff
@@ -560,16 +560,33 @@ weechat_lua_load (const char *filename, const char *code)
         return NULL;
     }
 
-#ifdef LUA_VERSION_NUM /* LUA_VERSION_NUM is defined only in lua >= 5.1.0 */
-    luaL_openlibs (lua_current_interpreter);
-#else
-    luaopen_base (lua_current_interpreter);
-    luaopen_string (lua_current_interpreter);
-    luaopen_table (lua_current_interpreter);
-    luaopen_math (lua_current_interpreter);
-    luaopen_io (lua_current_interpreter);
-    luaopen_debug (lua_current_interpreter);
-#endif /* LUA_VERSION_NUM */
+    // Lua 5.0.x libraries
+    luaopen_base      (lua_current_interpreter);
+    luaopen_io        (lua_current_interpreter);
+    luaopen_string    (lua_current_interpreter);
+    luaopen_table     (lua_current_interpreter);
+    luaopen_math      (lua_current_interpreter);
+    luaopen_os        (lua_current_interpreter);
+
+#ifdef LUA_VERSION_NUM
+#if   LUA_VERSION_NUM == 501 // Lua 5.1.x
+    luaopen_package   (lua_current_interpreter);
```

Comment:
Would it be possible to have something cumulative here?
That means: if Lua version ≥ 5.1.x, call `luaopen_package`, so this is done only one time.
Then if Lua version ≥ 5.2.x, call extra functions, etc.

Reply:
It behaves like a switch case, since `LUA_VERSION_NUM` can only be equal to one number at a time.

Reply:
Reverted, now if `LUA_VERSION_NUM` is NOT defined (compiling against Lua 5.0) it calls each `luaopen_*` function separately, since `luaL_openlibs` does not exist on that Lua version.

If it is defined, it automatically opens all libraries trough `luaL_openlibs`.

Reply:
> Would it be possible to have something cumulative here?
> That means: if Lua version ≥ 5.1.x, call `luaopen_package`, so this is done only one time.
> Then if Lua version ≥ 5.2.x, call extra functions, etc.

I just noticed i missed your question, sorry!

It would require additional checks, for example:
Lua 5.3 and Lua 5.2 have the bit32 library.
Lua 5.3 added built-in bitwise operators, deprecating the `bit32` library.
Lua 5.4 fully removes the `bit32` library.

Which means something like this would be needed:

```c
(...)
#elif LUA_VERSION_NUM >= 502
#if LUA_VERSION_NUM != 504
    luaopen_bit32 (lua_current_interpreter);
#endif
#elif (...)
```